test/timezones/Olsen.jl

```diff
@@ -0,0 +1,201 @@
+zone["WSDT"] = FixedTimeZone("WSDT", 46800, 3600)
+
+@test apia.transitions[1] == Transition(DateTime(1800,1,1), zone["LMT_OLD"])
+@test apia.transitions[2] == Transition(DateTime(1879,7,4,11,26,56), zone["LMT"])
+@test apia.transitions[3] == Transition(DateTime(1911,1,1,11,26,56), zone["WSST_OLD"])
+@test apia.transitions[4] == Transition(DateTime(1950,1,1,11,30), zone["SST"])
+@test apia.transitions[5] == Transition(DateTime(2010,9,26,11), zone["SDT"])
+@test apia.transitions[6] == Transition(DateTime(2011,4,2,14), zone["SST"])
+@test apia.transitions[7] == Transition(DateTime(2011,9,24,14), zone["SDT"])
+@test apia.transitions[8] == Transition(DateTime(2011,12,30,10), zone["WSDT"])
+@test apia.transitions[9] == Transition(DateTime(2012,3,31,14), zone["WSST"])
+@test apia.transitions[10] == Transition(DateTime(2012,9,29,14), zone["WSDT"])
+
+
+# Behaviour of mixing "RULES" as a String and as a Time. In reality this behaviour has never
+# been observed.
+
+# Manually generate zones and rules as if we had read them from a file.
+zones = ZoneDict()
+rules = RuleDict()
+
+zones["Pacific/Test"] = [
+    zoneparse("-10:00", "-", "TST-1", "1933 Apr 1 2:00s"),
+    zoneparse("-10:00", "1:00", "TDT-2", "1933 Sep 1 2:00s"),
+    zoneparse("-10:00", "Testing", "T%sT-3", "1934 Sep 1 3:00s"),
+    zoneparse("-10:00", "1:00", "TDT-4", "1935 Sep 1 3:00s"),
+    zoneparse("-10:00", "Testing", "T%sT-5", ""),
+]
+rules["Testing"] = [
+    ruleparse("1934", "1935", "-", "Apr", "1", "3:00s", "1", "D"),
+    ruleparse("1934", "1935", "-", "Sep", "1", "3:00s", "0", "S"),
+]
+
+test = resolve("Pacific/Test", zones, rules)
+
+zone = Dict{String,FixedTimeZone}()
+zone["TST-1"] = FixedTimeZone("TST-1", -36000, 0)
+zone["TDT-2"] = FixedTimeZone("TDT-2", -36000, 3600)
+zone["TST-3"] = FixedTimeZone("TST-3", -36000, 0)
+zone["TDT-3"] = FixedTimeZone("TDT-3", -36000, 3600)
+zone["TDT-4"] = FixedTimeZone("TDT-4", -36000, 3600)
+zone["TST-5"] = FixedTimeZone("TST-5", -36000, 0)
+zone["TDT-5"] = FixedTimeZone("TDT-5", -36000, 3600)
+
+@test test.transitions[1] == Transition(DateTime(1800,1,1), zone["TST-1"])
+@test test.transitions[2] == Transition(DateTime(1933,4,1,12), zone["TDT-2"]) # -09:00
+@test test.transitions[3] == Transition(DateTime(1933,9,1,12), zone["TST-3"])
+@test test.transitions[4] == Transition(DateTime(1934,4,1,13), zone["TDT-3"])
+@test test.transitions[5] == Transition(DateTime(1934,9,1,13), zone["TDT-4"])
+@test test.transitions[6] == Transition(DateTime(1935,9,1,13), zone["TST-5"])
+
+# Note: Due to how I wrote the zones/rules a duplicate transition exists. The TimeZone code
+# should be able to safely handle this but nothing should require duplicates.
+@test test.transitions[6] == test.transitions[7]
+
+
+# Make sure that we can deal with Links. Take note that the current implementation converts
+# links into zones which makes it hard to explicitly test for a link. We expect that the
+# following link exists:
```

Comment:
Links just copy the `Zone` type. The `Rule`s get reused. We could change the code such that `resolve` also takes in links and only performs the copy at the last possible moment. I didn't feel that the minor decrease in memory was worth the restructure.